pkg/reconciler/managed/reconciler.go

```diff
@@ -662,6 +665,21 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"external-name", meta.GetExternalName(managed),
 	)
 
+	// check the pause annotation and return if it has the value "true"
+	// after logging, publishing an event and updating the SYNC status condition
+	if val, ok := managed.GetAnnotations()[meta.AnnotationKeyReconciliationPaused]; ok && val == "true" {
+		log.Debug("Reconciliation is paused via the pause annotation", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
+		record.Event(managed, event.Normal(reasonReconciliationPaused, "Reconciliation is paused via the pause annotation"))
+		managed.SetConditions(xpv1.ReconcilePaused())
+		// if the pause annotation is removed, we will have a chance to reconcile again and resume
+		// and if status update fails, we will reconcile again to retry to update the status
+		return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
+	}
+	if managed.GetCondition(xpv1.TypeSynced).Reason == xpv1.ReasonReconcilePaused {
+		log.Debug("Reconciliation resumes")
+		record.Event(managed, event.Normal(reasonReconciliationResumes, "Reconciliation resumes"))
+	}
```

Comment:
It's possible these could be emitted multiple times, e.g. if the following logic encountered an error before updating the synced condition. My preference would be to just omit this block - I think it would be pretty easy to infer that the resource was unpaused when it started emitting other events/statuses (create, update, etc).